### PR TITLE
Split windows.py into a folder with separate modules

### PR DESCRIPTION
**Pull request by mcdo0486**
Splits windows.py into a folder with separate modules.
This will make it easier to develop new window classes outside of a monolithic module.

### CODE REVIEW

**Comment by msmttchr** [comment]
> Splits windows.py into a folder with separate modules. This will make it easier to develop new window classes outside of a monolithic module.

Thanks again for your effort and contribution.
I think it's best to wait outcome of #594 before reviewing this, since the principle is the same. Just a small comment for the time being, do you think it's worth splitting `ManagedImageWindow` in a separate file?

**Review by msmttchr — APPROVED** [review]
Ok for me. Possibly @CasperSchippers can double check.

**Comment by mcdo0486** [comment]
I fixed some documentation for widgets in a [598](https://github.com/pymeasure/pymeasure/pull/598). I'll do the same here once that has been merged because some links will need to be updated.

File: pymeasure/display/windows/image_window.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from ..widgets import (
+    ImageWidget,
+)
+from .managed_window import ManagedWindow
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class ManagedImageWindow(ManagedWindow):
```
Comment (CasperSchippers) [inline pymeasure/display/windows/image_window.py:39]: I'd be inclined to name this file managed_image_window.py, seems a bit more clear/consistent to me.

File: pymeasure/display/windows/managed_window.py [hunk 2]
```diff
@@ -630,7 +537,8 @@ def directory(self, value):
 
 class ManagedWindow(ManagedWindowBase):
```
Comment (CasperSchippers) [inline pymeasure/display/windows/managed_window.py:538]: Just a thought; would it not also make sense to split this file into one for ManagedWindowsBase and one for ManagedWindow?

**Review by CasperSchippers — COMMENTED** [review]
Left a few comments, mostly looks good to me.

File: pymeasure/display/windows/managed_window.py (hunk 2 above)
Comment (bilderbuchi) [reply pymeasure/display/windows/managed_window.py:538]: I'd say multiple classes per module is definitely fine in python, and imo having every class in a separate module does not improve the code structure.

**Comment by mcdo0486** [comment]
> I fixed some documentation for widgets in a [598](https://github.com/pymeasure/pymeasure/pull/598). I'll do the same here once that has been merged because some links will need to be updated.

Thanks for merging the other pull request. I've updated the various widget docstrings to be consistent with the refactor in this pull request.

File: pymeasure/display/windows/image_window.py (hunk 1 above)
Comment (mcdo0486) [reply pymeasure/display/windows/image_window.py]: Good point, I renamed the file.

File: pymeasure/display/windows/managed_window.py [hunk 3]
```diff
@@ -1,16 +1,3 @@
...
-# Copyright (c) 2013-2022 PyMeasure Developers
-#
-# Permission is hereby granted, free of charge, to any person obtaining a copy
-# of this software and associated documentation files (the "Software"), to deal
-# in the Software without restriction, including without limitation the rights
-# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-# copies of the Software, and to permit persons to whom the Software is
-# furnished to do so, subject to the following conditions:
-#
-# The above copyright notice and this permission notice shall be included in
```
[the remark this answers is not in the document]
Comment (mcdo0486) [reply pymeasure/display/windows/managed_window.py]: I fixed that section.

**Comment by bilderbuchi** [comment]
@CasperSchippers good to merge?

**Review by CasperSchippers — APPROVED** [review]
Yes, looks good to me.
Thanks for the work and sorry for the long wait.

**Pull request merged**